htlcswitch/link.go

```diff
@@ -1190,6 +1199,37 @@ func (l *channelLink) htlcManager() {
 		}
 	}
 
+	// If a shutdown message has previously been sent on this link, then we
+	// need to make sure that we have disabled any HTLC adds on the outgoing
+	// direction of the link and that we re-resend the same shutdown message
+	// that we previously sent.
+	l.cfg.ShutdownMsg.WhenSome(func(shutdown lnwire.Shutdown) {
+		// Immediately disallow any new outgoing HTLCs.
+		if !l.DisableAdds(Outgoing) {
+			l.log.Warnf("Outgoing link adds already disabled")
+		}
+
+		sendShutdown := func() {
+			err := l.cfg.Peer.SendMessage(false, &shutdown)
+			if err != nil {
+				l.log.Warnf("Error sending shutdown "+
+					"message: %v", err)
+			}
+		}
+
+		// If we still owe the peer a CommitSig message, then we queue
```

Comment:
but `htlcManager` is spun of in a goroutine right? so isnt there a chance that the switch has already lined up some new adds since `NotifyActiveLink` has already been called above?..

hmm so maybe `l.DisableAdds(Outgoing)` should even be done before `NotifyActiveLink`...?

Reply:
sorry I wasn't clear, I meant removing the `OweCommitment` bit. The switch won't queue up adds because `IsEligibleToForward` will return false since `markReestablished` hasn't been called yet so the placement of `DisableAdds` is ok

Reply:
gotcha! 🙏

Reply:
updated